server/types/src/game.rs

```diff
@@ -56,6 +57,10 @@ pub struct Game {
     pub time: Time,
     /// The event handler map.
     pub event_handlers: Arc<EventHandlers>,
+    // The event scheduler
+    // ToDo: If this struct is accessible from multiple threads,
+    // shouldn't this use a Mutex?
+    pub event_scheduler: EventScheduler,
```

Comment:
What would be the best way to implement such an event scheduler?

Reply:
For timing redstone, cloks and etc? If so, you might want to implement tile ticks instead and just use a system to advance the tile ticks.

Reply:
That is basically what I have implemented here. What I was wondering is if something like a mutext is neccessary here. Sorry for the confusion, as I am currently learing rust I am not sure what the appropriate way is to store this event scheduler.

Reply:
It would probably best to tie those tile ticks in with the current block update system used for creating falling blocks and the block support checks right?
When a block is changed we notify the neighbouring blocks by creating block update entities which are handled by other systems to actually do the required work. I think it would make sense to add a property to those entities to track in how many ticks or at what specific tick they should be processed and only act on them in the corresponding system when that time is reached.

Reply:
Ahh ok, then it should be fine and no need for Mutex; take a look at how chunk entities are handled, they are very similar.